lib/src/solver/package_lister.dart

```diff
@@ -259,9 +270,15 @@ class PackageLister {
         if (id.isRoot)
           ...pubspec.devDependencies.values
               .where((range) => !_overriddenPackages.contains(range.name)),
+        if (id.isRoot)
```

Comment:
Maybe include this in the condition above? (I would try it out to see if it gives a better overview when reading the code. If you finde that it makes it clumsy just keep it as it is)

Reply:
Done